algorithms/ar-soer/java/SieveOfEratosthenes.java

```diff
@@ -0,0 +1,43 @@
+//Sieve of Eratosthenes
+
+
+class SieveOfEratosthenes 
+{ 
+	void sieveOfEratosthenes(int number) 
+	{ 
+		
+		boolean prime[] = new boolean[number+1]; 
+		for(int preSet=0;preSet<number;preSet++) 
+			prime[preSet] = true; 
+		
+		for(int checkNo = 2; checkNo*checkNo <=number; checkNo++) 
+		{ 
+			
+			if(prime[checkNo] == true) 
+			{ 
+				
+				for(int mul = checkNo*checkNo; mul <= number; mul += checkNo) 
+					prime[mul] = false; 
+			} 
+		} 
+		
+		
+		for(int primeNum = 2;primeNum <= number; primeNum++) 
+		{ 
+			if(prime[primeNum] == true) 
```

Comment:
Codacy found an issue: [Avoid unnecessary comparisons in boolean expressions](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6371972)